nntrainer/src/util_func.cpp

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (C) 2020 Samsung Electronics Co., Ltd. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @file	util_func.cpp
+ * @date	08 April 2020
+ * @brief	This is collection of math functions
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Jijoong Moon <jijoong.moon@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ */
+
+#ifndef __UTIL_FUNC_H__
+#define __UTIL_FUNC_H__
+#include "util_func.h"
+#include <assert.h>
+#include "math.h"
+#include "tensor.h"
+
+Tensors::Tensor softmaxPrime(Tensors::Tensor x) {
+  int batch = x.getBatch();
+  int width = x.getWidth();
+  int height = x.getHeight();
+  assert(height == 1);
```

Comment:
Minor comment: If the height is always 1, I think you don't need to keep `height` in the below loops (for readability).

Reply:
Yes. But this is just temporary. Once TensorDim Class is implemented successfully, then this alert will be deleted.

Reply:
Got it